### DISCUSSION TITLE
Custom Plugin Lib env variables from file

### DISCUSSION THREAD

**Discussion by fabio-percivaldi** [mia-platform/community#178] [Q&A] [answered]
We created a new service using the custom-plugin-lib version 5.1.5, from the changelog it's possible to see that there is a bug involving the environment variables that has been fixed with the 5.1.5 itself but even with the latest version we encounter a problem at the start up.
It seems that the environment variable file (default.env or local.env) is not read from the service.
<img width="1203" alt="image" src="https://user-images.githubusercontent.com/30175434/214051058-e69d0120-be36-4187-a140-f1fb56b4d134.png">

i was not able to find the solution in the changelog of the custom plugin or lc39, do you know what could be the problem?

Changing the version to `4.3.2` solves the problem.

**Comment by davidebianchi** [comment, marked as answer]
Hi @fabio-percivaldi, this is an issue more related to the custom-plugin-lib. I suggest to you to open an issue directly on the custom-plugin-lib repo (or also to transfer this discussion if possible).
If you set to source env to the command line, it works? I don't think that the auto read of the .env file it's enabled.
Something like:

```sh
set -a && source .env
```

**Reply by fabio-percivaldi** [reply]
Thanks @davidebianchi i moved the conversation [here](https://github.com/mia-platform/custom-plugin-lib/issues/326) on the custom plugin lib issues
